spec/requests/providers/means/cash_outgoings_controller_spec.rb

```diff
@@ -0,0 +1,239 @@
+          legal_aid_application.update!(provider_received_citizen_consent: true)
+        end
+
+        context "with income categories" do
+          let(:income_types) { create_list :transaction_type, 3, :credit_with_standard_name }
+          let!(:legal_aid_application) do
+            create :legal_aid_application, :with_applicant,
+                   :with_non_passported_state_machine, :applicant_entering_means, transaction_types: income_types
+          end
+
+          it "redirects to income summary page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_income_summary_index_path(legal_aid_application))
+          end
+        end
+
+        context "with outgoings categories but no income categories" do
+          let(:outgoings_categories) { create_list :transaction_type, 3, :debit_with_standard_name }
+          let(:legal_aid_application) do
+            create :legal_aid_application,
+                   :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types: outgoings_categories
+          end
+
+          it "redirects to the outgoings summary page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_outgoings_summary_index_path(legal_aid_application))
+          end
+        end
+
+        context "without income or outgoings categories" do
+          let(:params) { nothing_selected_params }
+          let(:legal_aid_application) do
+            create :legal_aid_application,
+                   :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types: []
+          end
+
+          it "redirects to the has dependants page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_means_has_dependants_path(legal_aid_application))
+          end
+        end
+      end
+
+      context "with bank statement upload flow" do
+        before do
+          legal_aid_application.provider.permissions << Permission.find_or_create_by(role: "application.non_passported.bank_statement_upload.*")
+          legal_aid_application.update!(provider_received_citizen_consent: false)
+        end
+
+        context "with income and outgoing categories" do
+          let(:legal_aid_application) do
+            create(:legal_aid_application, :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types:)
+          end
+
+          let(:transaction_types) { [create(:transaction_type, :credit_with_standard_name), create(:transaction_type, :debit_with_standard_name)] }
+
+          it "redirects to has dependants page" do
+            request
```

Comment:
NOT A BLOCKER: duplicated `request`s inside the `with bank statement upload flow` could be pulled out to the before block

Reply:
i personally prefer to have the repetition rather than a `before { request }` type block in the context as it would allow a new test of `expect { request }.to ..blah`. I have seen situations where the before is forgotten and double requests end up being inadvertently made.